test/e2e/e2e.go

```diff
@@ -175,9 +63,6 @@ var _ = ginkgo.SynchronizedAfterSuite(func() {
 			e2elog.Logf("Error gathering metrics: %v", err)
 		}
 	}
-	if framework.TestContext.NodeKiller.Enabled {
```

Comment:
Was it intentional to move this check into `framework.CleanupSuite`?

Reply:
Ah I see, it's to remove the dependency. to framework from `test/e2e/e2e.go`?

Reply:
@andrewsykim it was, but I see it wouldn't work. I should move it back here alongside with 
```	
if TestContext.NodeKiller.Enabled {
	nodeKiller := NewNodeKiller(TestContext.NodeKiller, c, TestContext.Provider)
	nodeKillerStopCh = make(chan struct{})
	go nodeKiller.Run(nodeKillerStopCh)
``` 
and keep the `nodeKillerStopCh` around this function.

Reply:
Yeah sounds good, would prefer to keep this PR scoped to refactoring and change any actual logic in a separate PR